### DISCUSSION TITLE
Feasibility of pure forward ray tracing

### DISCUSSION THREAD

**Discussion by jantma2** [mitsuba-renderer/mitsuba3#800] [General]
Hello,

is it already possible to perform a pur forward ray tracing (light source to camera) with Mitsuba?
I'm dealing with those ray tracers during my professional life all day and I wonder whether I could do that for hobby at home also?

In case the answer is 'yes', can Mitsuba handle multiple cameras during a single computation process?

Thanks for support.
BR

**Comment by njroussel** [comment]
Hi @jantma2 

This is indeed possible. In fact, the [`ptracer`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_integrators.html#particle-tracer-ptracer)  is an existing integrator plugin that starts tracing rays from light sources.

The "main" rendering interface is designed to only accept a single `Sensor`. However, we have implemented a [`batch`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_sensors.html#batch-sensor-batch) sensor which groups multiple sensors into a single object.
Also, it's pretty easy to deviate from the main rendering interface (`mi.render`), you can quickly put together a custom pipeline by using the building blocks of Mitsuba itself. Take a look at [this tutorial](https://mitsuba.readthedocs.io/en/stable/src/rendering/scripting_renderer.html) for an example of a script that somewhat rebuilds a pipeline.
